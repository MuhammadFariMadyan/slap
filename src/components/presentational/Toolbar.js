import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

import Tool from "../presentational/Tool";

const Container = styled.div`
  width: 100%;
  display: flex;
  border-top: 1px solid #ddd;
  border-bottom: 1px solid #ddd;
  padding: 4px 24px;
  box-sizing: border-box;
`;

const Toolbar = ({ className, items }) => (
  <Container className={className}>
    {items.map((item, index) => (
      <Tool key={index} {...item} />
    ))}
  </Container>
);

Toolbar.propTypes = {
  className: PropTypes.string,
  items: PropTypes.array
};

export default Toolbar;
