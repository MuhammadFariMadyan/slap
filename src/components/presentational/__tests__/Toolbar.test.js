import React from "react";
import { shallow } from "enzyme";

import Toolbar from "../Toolbar";

jest.mock("../Tool", () => "Tool");

describe("Toolbar", () => {
  const Dummy = () => <div />;

  const setup = propOverrides => {
    const props = {
      items: [
        {
          icon: <Dummy name="first" />,
          onClick: jest.fn()
        },
        {
          icon: <Dummy name="second" />,
          onClick: jest.fn()
        },
        {
          icon: <Dummy name="third" />,
          onClick: jest.fn(),
          isActive: true
        }
      ],
      ...propOverrides
    };

    const wrapper = shallow(<Toolbar {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    props.items.forEach((item, index) => {
      wrapper
        .find("Tool")
        .at(index)
        .simulate("click");

      expect(item.onClick).toBeCalled();
    });
  });
});
