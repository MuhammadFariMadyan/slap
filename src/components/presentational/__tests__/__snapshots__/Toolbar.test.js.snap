// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should renders default 1`] = `
<styled.div>
  <Tool
    icon={
      <Dummy
        name="first"
      />
    }
    key="0"
    onClick={[Function]}
  />
  <Tool
    icon={
      <Dummy
        name="second"
      />
    }
    key="1"
    onClick={[Function]}
  />
  <Tool
    icon={
      <Dummy
        name="third"
      />
    }
    isActive={true}
    key="2"
    onClick={[Function]}
  />
</styled.div>
`;
